orchagent/aclorch.cpp

```diff
@@ -2001,21 +2042,38 @@ sai_object_id_t AclRulePacket::getRedirectObjectId(const string& redirect_value)
     try
     {
         NextHopKey nh(target);
-        if (!m_pAclOrch->m_neighOrch->hasNextHop(nh))
+        if (m_pAclOrch->m_neighOrch->hasNextHop(nh))
         {
-            SWSS_LOG_ERROR("ACL Redirect action target next hop ip: '%s' doesn't exist on the switch", nh.to_string().c_str());
-            return SAI_NULL_OBJECT_ID;
+            m_redirect_target_next_hop = target;
+            m_pAclOrch->m_neighOrch->increaseNextHopRefCount(nh);
+            return m_pAclOrch->m_neighOrch->getNextHopId(nh);
```

Comment:
Looks like logic is changed. Previously it returned NULL OBJECT if `hasnexthop `failed. Should we keep the old behaviour?

Reply:
This field can take many variants, it can be Port, Lag, NH IP, NHG and now Tunnel NH. 

Previous logic is incorrect as if get NH fails, it should try to check if it is one of other variants instead of returning  SAI_NULL_OBJECT_ID